instrumentation/github.com/astaxie/beego/beego.go

```diff
@@ -0,0 +1,152 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package beego
+
+import (
+	"context"
+	"net/http"
+
+	"google.golang.org/grpc/codes"
+
+	otelhttp "go.opentelemetry.io/contrib/instrumentation/net/http"
+	"go.opentelemetry.io/otel/api/trace"
+
+	"github.com/astaxie/beego"
+)
+
+// OTelBeegoHandler implements the http.Handler interface and provides
+// trace and metrics to beego web apps.
+type OTelBeegoHandler struct {
+	http.Handler
+}
+
+func (o *OTelBeegoHandler) ServeHTTP(rr http.ResponseWriter, req *http.Request) {
```

Comment:
Missing documentation. Possibly ...

```suggestion
// ServerHTTP calls the configured handler to serve HTTP for req to rr.
func (o *OTelBeegoHandler) ServeHTTP(rr http.ResponseWriter, req *http.Request) {
```